plugins/lookup/lookup.py

```diff
@@ -0,0 +1,205 @@
+            - NS
+            - NSEC
+            - NSEC3
+            - NSEC3PARAM
+            - PTR
+            - RP
+            - RRSIG
+            - SOA
+            - SPF
+            - SRV
+            - SSHFP
+            - TLSA
+            - TXT
+    query_retry:
+        description:
+            - Number of retries for DNS query timeouts.
+        type: int
+        default: 3
+    query_timeout:
+        description:
+            - Timeout per DNS query in seconds.
+        type: float
+        default: 10
+    server:
+        description:
+            - The DNS server(s) to use to look up the result.
+            - By default, the system's standard resolver is used.
+        type: list
+        elements: str
+    servfail_retries:
+        description:
+            - How often to retry on SERVFAIL errors.
+        type: int
+        default: 0
+    nxdomain_handling:
+        description:
+            - How to handle NXDOMAIN errors. These appear if an unknown domain name is queried.
+            - V(empty) (default) returns an empty result for that domain name.
+              This means that for the corresponding domain name, nothing is added to RV(_result).
+            - V(fail) makes the lookup fail.
+            - V(message) adds the string V(NXDOMAIN) to RV(_result).
+        type: str
+        choices:
+            - empty
+            - fail
+            - message
+        default: empty
+'''
+
+EXAMPLES = """
+- name: Look up A (IPv4) records for example.org
+  ansible.builtin.debug:
+    msg: "{{ lookup('community.dns.lookup', 'example.org.') }}"
+
+- name: Look up AAAA (IPv6) records for example.org
+  ansible.builtin.debug:
+    msg: "{{ lookup('community.dns.lookup', 'example.org.', type='AAAA' ) }}"
+"""
+
+RETURN = """
+_result:
+    description:
+        - The records of type O(type) for all queried DNS names.
+        - If multiple DNS names are queried in O(_terms), the resulting lists have been concatenated.
+    type: list
+    elements: str
+    sample:
+        - 127.0.0.1
+"""
+
+from ansible.errors import AnsibleLookupError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils.common.text.converters import to_text
+
+from ansible_collections.community.dns.plugins.module_utils.ips import (
+    is_ip_address,
+)
+
+from ansible_collections.community.dns.plugins.module_utils.dnspython_records import (
+    NAME_TO_RDTYPE,
+)
+
+from ansible_collections.community.dns.plugins.module_utils.resolver import (
+    SimpleResolver,
+)
+
+from ansible_collections.community.dns.plugins.plugin_utils.ips import (
+    assert_requirements_present as assert_requirements_present_ipaddress,
+)
+
+from ansible_collections.community.dns.plugins.plugin_utils.resolver import (
+    assert_requirements_present as assert_requirements_present_dnspython,
+    guarded_run,
+)
+
+try:
+    import dns.resolver
+except ImportError:
+    # handled by assert_requirements_present_dnspython
+    pass
```

Comment:
This is quite a large PR and I'm not sure if I understand everything. But I have a question about this: why do you do this test?

I think you've removed code from [plugins/modules/nameserver_info.py](https://github.com/ansible-collections/community.dns/pull/155/files#diff-757d8685603b472bf9b7b32f64dea9ee43ed1360bf6cee8404aa533b0950b315) which looks more or less the same, or at least very similar. Is there a reason to keep this test here? Although, it doesn't really test anything... if `dns.resolver` can't be imported, nothing happens. Well, something might fail later when the code tries to really use it but nothing happens _here_.

If you have this code here on purpose, I'm sure you have a very good reason (which I just don't see at the moment). On the other hand, I thought you might have overlooked this.

Reply:
It's needed here because `dns.resolver` is needed in this Python file (`except dns.resolver.NXDOMAIN:`), while dnspython is no longer (directly) needed in nameserver_info (the exception catching moved to the generic module_utils code).

Reply:
I've moved out some of the changes unrelated to the new plugins to #156, #157, and #158 (will create that one once 157 is merged). That should make this one somewhat simpler.